controllers/volumereplicationgroup_controller.go

```diff
@@ -1072,21 +1027,23 @@ func (v *VRGInstance) updateVRGStatus(updateConditions bool) error {
 			v.log.Info(fmt.Sprintf("Failed to update VRG status (%v/%+v)",
 				err, v.instance.Status))
 
-			return fmt.Errorf("failed to update VRG status (%s/%s)", v.instance.Name, v.instance.Namespace)
+			result.Requeue = true
+
+			return result
```

Comment:
Updating the vrg status is not related to to the controller result. Why we we passing unrelated argument just to return it?

Reply:
The idea here is to reduce the duplicated error handling that both callers of this routine do.  It makes sense to me to set the requeue flag when the error condition is encountered.  Pass and return by value was a style choice.  Alternatively, the requeue bool could have been passed by reference as done elsewhere.